packages/blockchain-extension/extension/commands/packageSmartContractCommand.ts

```diff
@@ -115,25 +116,37 @@ export async function packageSmartContract(workspace?: vscode.WorkspaceFolder, o
             // Determine the path argument.
             let contractPath: string = workspace.uri.fsPath; // Workspace path
             if (language === 'golang') {
-
                 if (!process.env.GOPATH) {
-                    // The path is relative to $GOPATH/src for Go smart contracts.
-                    const srcPath: string = path.join(contractPath, '..', '..', 'src');
-                    contractPath = path.basename(contractPath);
-                    const exists: boolean = await fs.pathExists(srcPath);
-
-                    if (!exists) {
-                        // Project path is not under GOPATH.
+                    // The path is relative to $GOPATH/src for Go smart contracts.
+                    const indexSrc: number = contractPath.indexOf(path.sep + 'src' + path.sep);
+                    if (indexSrc === -1) {
+                        // Project path is not under GOPATH.
                         throw new Error('The environment variable GOPATH has not been set, and the extension was not able to automatically detect the correct value. You cannot package a Go smart contract without setting the environment variable GOPATH.');
                     } else {
+                        const srcPath: string = contractPath.substring(0, indexSrc + 4);
+                        contractPath = path.relative(srcPath, contractPath);
                         process.env.GOPATH = path.join(srcPath, '..');
                     }
                 } else {
-                    // The path is relative to $GOPATH/src for Go smart contracts.
-                    const srcPath: string = path.join(process.env.GOPATH, 'src');
+                    // The path is relative to $GOPATH/src for Go smart contracts.
+                    const indexSrc: number = contractPath.indexOf(path.sep + 'src' + path.sep);
+                    if (indexSrc === -1) {
+                        throw new Error('The Go smart contract is not a subdirectory of the path specified by the environment variable GOPATH. Please correct the environment variable GOPATH.');
+                    }
+                    const srcPath: string = contractPath.substring(0, indexSrc + 4);
+                    const goPaths: string[] = process.env.GOPATH.split(path.delimiter);
+                    let pathsMatch: boolean = false;
+                    if (goPaths.length > 1) {
+                        originalGOPATH = process.env.GOPATH;
+                    }
+                    goPaths.forEach((value: string) => {
+                        if (value === srcPath.substr(0, srcPath.length - 4)) {
```

Comment:
This implies that the path stored in value does not end with a directory delimiter: `srcPath.substr(0, srcPath.length - 4)`

That is not necessarily true, so we should replace the last delimiter of value if present before comparing.